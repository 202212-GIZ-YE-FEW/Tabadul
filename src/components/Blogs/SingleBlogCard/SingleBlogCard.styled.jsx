import styled from "@emotion/styled";
import Image from "next/image";

export const Container = styled.div`
    background: #ffffff;
    box-shadow: 0px 0px 7px rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    width: 338px;
    height: 413px;
    transition: all 0.3s ease-in-out;
    &:hover {
        transform: translateY(-5px);
        box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
    }
`;

export const BlogCard = styled.div`
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    gap: 20px;
`;

export const ImageContainer = styled.div`
    padding-top: 4.68px;
    position: relative;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    gap: 20px;
`;

export const BlogImage = styled(Image)`
    border-radius: 19px 19px 0px 0px;
`;
export const CardDate = styled.div`
    position: absolute;
    left: 1.4%;
    bottom: 0%;
    width: 120.27px;
    line-height: 40.52px;
    text-align: center;
    color: #ffffff;
    background: #33956d;
`;
export const BlogTitle = styled.div`
    color: #3c4347;
    text-align: center;
    margin-top: 12.92px;
    font-weight: 700;
    font-size: 18px;
    line-height: 29px;
`;

export const BlogDescription = styled.p`
    color: #3c4347;
    font-size: 20px;
    line-height: 30px;
    text-align: center;
    padding: 10px 17.29px 0 17.29px;
    margin: 13.31px 0 29.61px 0;
`;
