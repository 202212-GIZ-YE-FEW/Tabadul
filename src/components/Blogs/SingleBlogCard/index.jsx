import React from "react";

import {
    BlogDescription,
    BlogImage,
    BlogTitle,
    CardDate,
    Container,
    ImageContainer,
} from "./SingleBlogCard.styled";

function SingleBlogCard(props) {
    const { blogTitle, blog, blogDate, blogImage } = props;

    const dateIntoArray = blogDate?.split("/");

    function getMonthName(monthNumber) {
        const date = new Date();
        date.setMonth(monthNumber - 1);

        return date.toLocaleString("en-US", {
            month: "long",
        });
    }

    return (
        <Container>
            <ImageContainer>
                <BlogImage
                    src={blogImage}
                    alt='Blog Image'
                    width={328.57}
                    height={217.41}
                />
                <CardDate>
                    {dateIntoArray ? dateIntoArray[0] : null}{" "}
                    {getMonthName(dateIntoArray ? dateIntoArray[1] : null)}
                </CardDate>
            </ImageContainer>
            <BlogTitle>{blogTitle}</BlogTitle>
            <BlogDescription>
                {blog?.length > 50 ? `${blog.substring(0, 50)} ...` : blog}
            </BlogDescription>
        </Container>
    );
}

export default SingleBlogCard;
