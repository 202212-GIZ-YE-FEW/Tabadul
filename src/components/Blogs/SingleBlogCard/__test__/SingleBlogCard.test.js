import renderer from "react-test-renderer";

import SingleBlogCard from "..";

it("renders correctly", () => {
    const tree = renderer.create(<SingleBlogCard />).toJSON();
    expect(tree).toMatchSnapshot();
});
