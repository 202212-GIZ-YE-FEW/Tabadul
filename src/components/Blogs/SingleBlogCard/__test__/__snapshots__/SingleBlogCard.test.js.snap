// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-1st69g8"
>
  <div
    className="css-1k7kckz"
  >
    <img
      alt="Blog Image"
      className="css-1ygfzho"
      data-nimg="1"
      decoding="async"
      height={217.41}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="images/donation.svg"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={328.57}
    />
    <div
      className="css-1xxa61n"
    >
      2 July
    </div>
  </div>
  <div
    className="css-qhrrsi"
  >
    Blog Title
  </div>
  <p
    className="css-s62muy"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
  </p>
</div>
`;
